File: android/library/src/main/kotlin/com/dongyuzhao/chatfeed/ChatFeed.kt
package com.dongyuzhao.chatfeed

data class ChatFeedMessage(
    val author: String,
    val message: String,
)

data class ChatFeed(
    val title: String,
    val messages: List<ChatFeedMessage>,
) {
    companion object {
        fun sample(): ChatFeed = ChatFeed(
            title = "Chat Feed Sample",
            messages = listOf(
                ChatFeedMessage(author = "System", message = "Welcome to Chat Feed."),
                ChatFeedMessage(author = "Taylor", message = "This sample app is backed by the Android library."),
            ),
        )
    }
}

File: swift/Sources/SwiftChatFeed/ChatFeed.swift
public struct ChatFeedItem: Sendable, Equatable {
    public let author: String
    public let message: String

    public init(author: String, message: String) {
        self.author = author
        self.message = message
    }
}

public struct ChatFeed: Sendable, Equatable {
    public let title: String
    public let items: [ChatFeedItem]

    public init(title: String = "Chat Feed", items: [ChatFeedItem] = []) {
        self.title = title
        self.items = items
    }

    public static var sample: ChatFeed {
        ChatFeed(
            title: "Chat Feed Sample",
            items: [
                ChatFeedItem(author: "System", message: "Welcome to Chat Feed."),
                ChatFeedItem(author: "Taylor", message: "This sample is wired to the Swift package."),
            ]
        )
    }
}
